quickwit/quickwit-indexing/src/source/ingest_api_source.rs

```diff
@@ -307,6 +307,8 @@ mod tests {
         let doc_batches: Vec<RawDocBatch> = doc_processor_inbox.drain_for_test_typed();
         assert_eq!(doc_batches.len(), 2);
         assert!(doc_batches[1].docs[0].starts_with("038462"));
+        ingest_api_source_handle.quit().await;
```

Comment:
was it necessary to quit ingest_api_source independently?
If justified, a comment would be helpful

Reply:
It was deadlocking occasionally in my tests. I still need to figure out where the deadlock was occurring but, indeed, I should've added a comment.